File: MyFarmProduce.Domain/Entities/ChatMessage.cs
namespace MyFarmProduce.Domain.Entities;

/// <summary>A message in the shared community chat room.</summary>
public class ChatMessage
{
    public int Id { get; set; }

    public int CustomerId { get; set; }
    public Customer? Customer { get; set; }

    public string SenderName { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: domain/src/main/java/com/myfarmproduce/domain/entity/ChatMessage.java
package com.myfarmproduce.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

/** A message in the shared community chat room. */
@Entity
@Table(name = "chat_messages")
@Getter
@Setter
public class ChatMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    private Customer customer;

    @Column(name = "sender_name", nullable = false, length = 150)
    private String senderName = "";

    @Column(nullable = false, length = 2000)
    private String content = "";

    @Column(name = "created_at", nullable = false)
    private Instant createdAt = Instant.now();
}
